src/Diagnostics.RuntimeHost/Services/DiagnosticTranslator/DiagnosticTranslatorService.cs

```diff
@@ -0,0 +1,288 @@
+        }
+
+        public async Task<DiagnosticData> GetInsightsTranslation(DiagnosticData dataset, string language)
+        {
+            int messageColumnIndex = 1;
+            int insightNameColumnIndex = 2;
+            int insightValueColumnIndex = 3;
+            int insightSolutionColumnIndex = 5;
+
+            return await GetRenderingTranslation(dataset, language, new List<int> { messageColumnIndex, insightNameColumnIndex, insightValueColumnIndex, insightSolutionColumnIndex });
+        }
+
+        // Cognitive translation API has a limit of 100 elements for string arrays to be sent in request body. 
+        private async Task<List<string>> GetGroupTranslations(List<string> textsToTranslate, string languageToTranslate)
+        {
+            int translationGroupCount = textsToTranslate.Count / 100;
+            int lastGroupSize = textsToTranslate.Count % 100;
+            List<List<string>> translationGroupList = new List<List<string>>();
+            for (int i = 0; i <= translationGroupCount; i++)
+            {
+                int groupSize = i == translationGroupCount ? lastGroupSize : 100;
+                translationGroupList.Add(textsToTranslate.GetRange(i * 100, groupSize));
+            }
+
+            var translatedTextsGroup = await Task.WhenAll(translationGroupList.Select(textBatch => GetTranslations(textBatch, languageToTranslate)));
+            return translatedTextsGroup.Where(group => group != null).SelectMany(group => group).ToList();
+        }
+
+        // Translation method to make the http request with string arrays to be translated
+        public async Task<List<string>> GetTranslations(List<string> textsToTranslate, string languageToTranslate)
+        {
+            if (textsToTranslate == null)
+            {
+                return null;
+            }
+
+            // Always look for cache before calling translator API
+            Tuple<string, string> key = new Tuple<string, string>(languageToTranslate, string.Join("_", textsToTranslate));
+            if (this._translationCacheService.TryGetValue(key, out List<string> value))
+            {
+                return value;
+            }
+
+            string route = string.Format("/translate?api-version=3.0&from=en&to={0}", languageToTranslate);
+            object[] body = textsToTranslate.Select((text) => { return new { Text = text }; }).ToArray();
+            var requestBody = JsonConvert.SerializeObject(body);
+
+            try
+            {
+                using (var request = new HttpRequestMessage())
+                {
+                    request.Method = HttpMethod.Post;
+                    request.RequestUri = new Uri(translatorBaseURL + route);
+                    request.Content = new StringContent(requestBody, Encoding.UTF8, "application/json");
+
+                    HttpResponseMessage response = await _httpClient.SendAsync(request).ConfigureAwait(false);
```

Comment:
You can remove `ConfigureAwait(false)` and anywhere else you might be doing this.